src/jose/Jws.ts

```diff
@@ -0,0 +1,132 @@
+import * as ed25519 from '@noble/ed25519';
+import * as secp256k1 from '@noble/secp256k1';
+import base64url from 'base64url';
+import type JwkEd25519Private from './JwkEd25519Private';
+import type JwkEd25519Public from './JwkEd25519Public';
+import type JwsFlattened from './JwsFlattened';
+import type JwkPrivate from './JwkPrivate';
+import type JwkPublic from './JwkPublic';
+import type JwkSecp256k1Private from './JwkSecp256k1Private';
+import type JwkSecp256k1Public from './JwkSecp256k1Public';
+
+/**
+ * Class containing reusable JWS operations.
+ */
+export default class Jws {
+  /**
+   * Signs the given payload as a JWS.
+   * NOTE: this is mainly used by tests to create valid test data.
+   * @throws {Error} if key given is unsupported.
+   */
+  public static async sign (
+    protectedHeader: object,
+    payload: Buffer,
+    privateKeyJwk: JwkPrivate
+  ): Promise<JwsFlattened> {
+
+    const protectedHeaderString = JSON.stringify(protectedHeader);
+    const protectedHeaderBase64UrlString = base64url.encode(protectedHeaderString);
+    const payloadBase64UrlString = base64url.encode(payload);
+    const signingInputBase64urlString = protectedHeaderBase64UrlString + '.' + payloadBase64UrlString;
+    const signingInputBuffer = Buffer.from(signingInputBase64urlString);
+
+    // This is where we will add support for different algorithms over time.
+    let signatureBuffer: Buffer;
+    if (privateKeyJwk.crv === 'Ed25519') {
+      signatureBuffer = await Jws.signEd25519(signingInputBuffer, privateKeyJwk as JwkEd25519Private);
+    } else if (privateKeyJwk.crv === 'secp256k1') {
+      signatureBuffer = await Jws.signSecp256k1(signingInputBuffer, privateKeyJwk as JwkSecp256k1Private);
```

Comment:
```suggestion
      signatureBuffer = await Jws.signSecp256k1(signingInputBuffer, privateKeyJwk);
```
the as cast can be removed because we used string literals for `kty` and `crv`